src/Kestrel.Core/Internal/Http/SendFile.Windows.cs

```diff
@@ -0,0 +1,434 @@
+            {
+                var span = _memory.Span;
+                span = span.Slice(numberOfBytesRead);
+                _endChunkBytes.AsSpan().CopyTo(span);
+                _writer.Advance(numberOfBytesRead + _endChunkBytes.Count + _chunkHeaderLength);
+            }
+
+            public void Dispose()
+            {
+                if (!_isDisposed)
+                {
+                    if (_fileHandle != null && !_fileHandle.IsInvalid)
+                    {
+                        _fileHandle.Dispose();
+                    }
+                    _threadPoolBoundHandle?.Dispose();
+                    _preAllocatedOverlapped?.Dispose();
+                    _isDisposed = true;
+                }
+            }
+        }
+
+        private static int AvailableToWriteChunked(int totalRemaining, int totalAvailable)
+        {
+            if (totalAvailable - totalRemaining > 12) return totalRemaining;
+            // Bit lazy on the boundaries
+            var remaining = totalRemaining - 4; // remove the chunking delimeters
+            if (remaining > 0xfff_ffff)
+            {
+                remaining -= 8;
+            }
+            else if (remaining > 0xff_ffff)
+            {
+                remaining -= 7;
+            }
+            else if (remaining > 0xf_ffff)
+            {
+                remaining -= 6;
+            }
+            else if (remaining > 0xffff)
+            {
+                remaining -= 5;
+            }
+            else if (remaining > 0xfff)
+            {
+                remaining -= 4;
+            }
+            else if (remaining > 0xff)
+            {
+                remaining -= 3;
+            }
+            else if (remaining > 0xf)
+            {
+                remaining -= 2;
+            }
+            else
+            {
+                remaining -= 1;
+            }
+
+            return remaining;
+        }
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern unsafe bool SetFileCompletionNotificationModes(
+                SafeFileHandle file,
+                CompletionNotificationModes fileAccess
+            );
+
+        [DllImport("kernel32.dll", SetLastError = true)]
+        static extern unsafe int ReadFile(
+                SafeFileHandle file,
+                ref byte destination,
+                int bytesToRead,
+                out int bytesRead,
+                NativeOverlapped* overlapped
+            );
+
+        [DllImport("Kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
+        static extern SafeFileHandle CreateFile(
+            string fileName,
+            [MarshalAs(UnmanagedType.U4)] FileAccess fileAccess,
+            [MarshalAs(UnmanagedType.U4)] FileShare fileShare,
+            IntPtr securityAttributes,
+            [MarshalAs(UnmanagedType.U4)] FileMode creationDisposition,
+            [MarshalAs(UnmanagedType.U4)] FileAttributes flags,
+            IntPtr template
+        );
```

Comment:
Why make this windows-only for just these three P/Invokes? Is it really that much faster than the System.IO equivalents?